### PR TITLE
added test for tycho + xtend

### PR DESCRIPTION
**Pull request by cdietrich**
added test for tycho + xtend
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
have no idea how to test this. i also dont to test with 2.5.0 for the good case

File: tycho-its/projects/tycho.xtend/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>tycho.xtend</groupId>
+	<version>1.0.0-SNAPSHOT</version>
+	<artifactId>tycho.xtend</artifactId>
+	<packaging>eclipse-plugin</packaging>
+	
+	<properties>
```
Comment (cdietrich) [inline tycho-its/projects/tycho.xtend/pom.xml:9]: dont know what to keep in this pom and what to sort out

Comment (laeubi) [reply tycho-its/projects/tycho.xtend/pom.xml:9]: I think this is all fine.

**Comment by laeubi** [comment]
> have no idea how to test this. i also dont to test with 2.5.0 for the good case

You can run this on the commandline

`tycho/tycho-its$ mvn clean verify -Pits -Dtest=org.eclipse.tycho.test.xtend.TychoXtendTest`

in general `.project` and `.classpath` is not required, can you remove those, add the missing import as my change (sorry I committed it directly but actually tried to propose a change), and squash everything?

**Comment by laeubi** [comment]
@cdietrich I think this is the expected failure?

```
[DEBUG] Original compiler output: ----------
1. ERROR in /home/jenkins/agent/workspace/tycho-github_PR-464/tycho-its/target/projects/TychoXtendTest/projectA/tycho.xtend/src/demo/SomeJavaClass.java (at line 15)
	public class SomeJavaClass extends SomeXtendClass {
	                                   ^^^^^^^^^^^^^^
SomeXtendClass cannot be resolved to a type
----------
1 problem (1 error)
```

If yes, can you add the 

```
<plugin>
	<groupId>org.eclipse.tycho</groupId>
	<artifactId>tycho-compiler-plugin</artifactId>
	<version>${tycho.version}</version>
	<executions>
		<execution>
			<id>verify-classpath</id>
			<phase>validate</phase>
			<goals>
				<goal>validate-classpath</goal>
			</goals>
			<configuration>
			</configuration>
		</execution>
	</executions>
</plugin>
```

so we see that the failure is fixed with this?

**Comment by cdietrich** [comment]
yes it helps at least in my sample project.
still struggling with reproducing it with the it

**Comment by laeubi** [comment]
> still struggling with reproducing it with the it

I think the itest already perfectly reproduce the issue :+1:

**Comment by cdietrich** [comment]
ok. locally its green somehow.

**Comment by laeubi** [comment]
> ok. locally its green somehow.

Have you installed a recent tycho snapshot in your local repository before running?

**Review by laeubi — APPROVED** [review]
This looks fine now, lets see if the build fails as expected.

**Comment by cdietrich** [comment]
yes i did

**Comment by laeubi** [comment]
> yes i did

Strange but at least in the CI build (https://github.com/eclipse/tycho/runs/4678909729) it shows the error you would have expected?

**Comment by cdietrich** [comment]
now i see fail locally too. and the verify-classpath makes it green again

**Comment by laeubi** [comment]
Thanks for verification I'll take a look how we can fix this without additional configuration.

**Comment by laeubi** [comment]
@cdietrich thanks for this test-case, everything is fine now, please check with your build when new snapshot is available if this fixes your problems.

**Pull request merged**